tasks/spi/src/main/kotlin/AbstractDITask.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.tasks.spi
+
+import com.typesafe.config.ConfigFactory
+
+import org.eclipse.apoapsis.ortserver.config.ConfigManager
+
+import org.koin.core.Koin
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.get
+import org.koin.core.module.Module
+import org.koin.dsl.koinApplication
+import org.koin.dsl.module
+
+import org.slf4j.LoggerFactory
+
+/**
+ * An abstract base class for [Task] implementations that require dependency injection.
+ *
+ * A concrete implementation must provide a list of Koin modules to be taken into account. The class then starts an
+ * isolated Koin context containing these modules. It expects that one of the objects in the context implements the
+ * [Task] interface. It obtains this object and calls its [Task.execute] method.
+ *
+ * Using this class, it is possible quite easily to get access to various components and infrastructure of ORT Server,
+ * including the database repositories, or the transport components. When setting up the context for dependency
+ * injection, a [ConfigManager] instance is included as well; so all configuration settings are available to the task.
+ */
+abstract class AbstractDITask(
+    /** A name for this task. */
+    val name: String
+) : Task {
+    companion object {
+        private val logger = LoggerFactory.getLogger(AbstractDITask::class.java)
+
+        /** A default name to be used for tasks if no explicit name was provided. */
+        private const val DEFAULT_TASK_NAME = "Unnamed task"
+
+        /**
+         * Create a new instance of a [AbstractDITask] that depends on the given [modules]. Optionally, the task can be
+         * given a [name].
```

Comment:
Why is the task name optional?

Reply:
The original idea was to use the simple class name as fallback, but this did not work out, since the name is `null` for anonymous classes. I have made the task name now mandatory.